class-wxr-import-ui.php

```diff
@@ -0,0 +1,635 @@
+		} else {
+			esc_html_e( 'or assign posts to an existing user:', 'wordpress-importer' );
+		}
+
+		wp_dropdown_users( array(
+			'name' => sprintf( 'user_map[%d]', $index ),
+			'multi' => true,
+			'show_option_all' => __( '- Select -', 'wordpress-importer' )
+		));
+
+		printf(
+			'<input type="hidden" name="imported_authors[%d]" value="%s" />',
+			(int) $index,
+			esc_attr( $author['user_login'] )
+		);
+
+		// Keep the old ID for when we want to remap
+		if ( isset( $author['ID'] ) ) {
+			printf(
+				'<input type="hidden" name="imported_author_ids[%d]" value="%d" />',
+				(int) $index,
+				esc_attr( $author['ID'] )
+			);
+		}
+
+		if ( $supports_extras )
+			echo '</div>';
+	}
+
+	/**
+	 * Decide whether or not the importer should attempt to download attachment files.
+	 * Default is true, can be filtered via import_allow_fetch_attachments. The choice
+	 * made at the import options screen must also be true, false here hides that checkbox.
+	 *
+	 * @return bool True if downloading attachments is allowed
+	 */
+	protected function allow_fetch_attachments() {
+		return apply_filters( 'import_allow_fetch_attachments', true );
+	}
+
+	/**
+	 * Decide whether or not the importer is allowed to create users.
+	 * Default is true, can be filtered via import_allow_create_users
+	 *
+	 * @return bool True if creating users is allowed
+	 */
+	protected function allow_create_users() {
+		return apply_filters( 'import_allow_create_users', true );
+	}
+
+	/**
+	 * Map old author logins to local user IDs based on decisions made
+	 * in import options form. Can map to an existing user, create a new user
+	 * or falls back to the current user in case of error with either of the previous
+	 */
+	protected function get_author_mapping( $args ) {
+		if ( ! isset( $args['imported_authors'] ) ) {
+			return array(
+				'mapping'        => array(),
+				'slug_overrides' => array(),
+			);
+		}
+
+		$map        = isset( $args['user_map'] ) ? (array) $args['user_map'] : array();
+		$new_users  = isset( $args['user_new'] ) ? $args['user_new'] : array();
+		$old_ids    = isset( $args['imported_author_ids'] ) ? (array) $args['imported_author_ids'] : array();
+
+		// Store the actual map.
+		$mapping = array();
+		$slug_overrides = array();
+
+		foreach ( (array) $args['imported_authors'] as $i => $old_login ) {
+			$old_id = isset( $old_ids[$i] ) ? (int) $old_ids[$i] : false;
+
+			if ( isset( $map[$i] ) ) {
+				$user = get_user_by( 'id', (int) $map[$i] );
+
+				if ( isset( $user->ID ) ) {
+					$mapping[] = array(
+						'old_slug' => $old_login,
+						'old_id'   => $old_id,
+						'new_id'   => $user->ID,
+					);
+				}
+			} elseif ( isset( $new_users[ $i ] ) ) {
+				if ( $new_users[ $i ] !== $old_login ) {
+					$slug_overrides[ $old_login ] = $new_users[ $i ];
+				}
+			}
+		}
+
+		return compact( 'mapping', 'slug_overrides' );
+	}
+
+	protected function emit_sse_message( $data ) {
```

Comment:
Maybe docblock this and the below methods, guessing not crucial though so not a blocker